guides/testing.md

```diff
@@ -1,8 +1,32 @@
 # Testing Crystal Code
 
-Crystal comes with a full-feature spec library to write test cases in the [`Spec`module](https://crystal-lang.org/api/latest/Spec.html).
+Crystal comes with a full-feature spec library in the [`Spec`module](https://crystal-lang.org/api/latest/Spec.html). It provides the structure for writing executable examples of how your code should behave.
 
-It is inspired by [Rspec](http://rspec.info/).
+Inspired by [Rspec](http://rspec.info/) it includes a domain specific language (DSL) that allows to write examples in a way similar to plain english..
+
+A basic spec looks something like this:
+
+```crystal
+require "spec"
+
+describe "Array" do
```

Comment:
I think we could mention it as `describe Array do ...` (without the quotes), as it's usually used like that!

Reply:
I's possible to use a constant, but there is no real gain from it. This is different in rspec, where the first argument to `describe` is - in case of a class - available as `described_class` and an instance of that class is created as [implicit subject](https://relishapp.com/rspec/rspec-core/v/3-0/docs/subject/implicitly-defined-subject).

In Crystal, it is just a a descriptor, the constant get's converted to `String` anyway. So, I'm not sure it should if we should promote this.

Reply:
I agree, but I still think that we should encourage using it as a Type, without quotes.

It's not really a _problem_ to have a typo or a mistake in an `it` description, but when it's a type description, I would like to have an error if the dev wrote `Arary` instead of `Array` (writing `"Arary"` with quotes won't give an error, and could be mis-leading).

We could say the same for the methods, where we usually write `describe "#empty?"` (with quotes), but as there's currently no way to check for the method existance, I think we can leave that out.